x/evm/types/utils.go

```diff
@@ -52,23 +58,33 @@ func rlpHash(x interface{}) (hash ethcmn.Hash) {
 	return hash
 }
 
+// ResultData represents the data returned in an comsos-sdk/types.Result
+type ResultData struct {
+	Address ethcmn.Address
+	Bloom   ethtypes.Bloom
+	Logs    []*ethtypes.Log
+	Ret     []byte
+}
+
 // EncodeReturnData takes all of the necessary data from the EVM execution
 // and returns the data as a byte slice
-func EncodeReturnData(addr ethcmn.Address, bloom ethtypes.Bloom, evmRet []byte) []byte {
-	// Append address, bloom, evm return bytes in that order
-	returnData := append(addr.Bytes(), bloom.Bytes()...)
-	return append(returnData, evmRet...)
+func EncodeResultData(data *ResultData) ([]byte, error) {
+	return rlp.EncodeToBytes(data)
 }
 
 // DecodeReturnData decodes the byte slice of values to their respective types
-func DecodeReturnData(bytes []byte) (addr ethcmn.Address, bloom ethtypes.Bloom, ret []byte, err error) {
-	if len(bytes) >= returnIdx {
-		addr = ethcmn.BytesToAddress(bytes[:bloomIdx])
-		bloom = ethtypes.BytesToBloom(bytes[bloomIdx:returnIdx])
-		ret = bytes[returnIdx:]
-	} else {
-		err = fmt.Errorf("Invalid format for encoded data, message must be an EVM state transition")
-	}
+func DecodeResultData(in []byte) (*ResultData, error) {
```

Comment:
why do you need to return the pointer?

Reply:
I figured pointers to structs are the usual practice in Go, but I can return the value instead, is there some sort of standard or guideline for this project in terms of Go patterns?

Reply:
I use pointers for the function params to update the type in place and avoid returning the value. Returning the pointer doesn't make much sense for that reason, i.e you return the value and then pass the pointer of the `ResultData`.

```go
func DecodeResultData(in []byte) (ResultData, error) {
```

and then

```go
resData, err := DecodeResultData(bz)
... 
someFunc(&resData) // to modify the result if necessary. If you don't update any value just pass the value
```

Reply:
makes sense, will update!